File: 0153-find-minimum-in-rotated-sorted-array/0153-find-minimum-in-rotated-sorted-array.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var findMin = function(nums) {
    let st = 0, en = nums.length -1;

    while(st < en){
        let mid = Math.floor(st + ( en - st) / 2);

        if(nums[mid] <= nums[en]) en = mid;
        else st = mid + 1;
    }
    return nums[st];
};

File: 0153-find-minimum-in-rotated-sorted-array/0153-find-minimum-in-rotated-sorted-array.cpp
class Solution {
public:
    int findMin(vector<int>& arr) {
        int st = 0, en = arr.size() - 1;

        while(st < en){
            int mid = st  + (en - st) / 2;
           if(arr[mid] <= arr[en]) en = mid;
           else st = mid + 1;
        }
        return arr[st];
    }
};